modules/programs/translate-shell.nix

```diff
@@ -0,0 +1,54 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.translate-shell;
+
+  mkKeyValue = key: value:
+    let
+      formatValue = v:
+        if isBool v then
+          (if v then "true" else "false")
+        else if isString v then
+          ''"${v}"''
+        else if isList v then
+          "[ ${concatStringsSep " " (map formatValue v)} ]"
+        else
+          toString v;
+    in ":${key} ${formatValue value}";
+
+  toKeyValue = generators.toKeyValue { inherit mkKeyValue; };
+
+in {
+  meta.maintainers = [ maintainers.marsam ];
+
+  options.programs.translate-shell = {
+    enable = mkEnableOption "translate-shell";
+
+    settings = mkOption {
+      type = with types; attrsOf (oneOf [ bool str (listOf str) ]);
+      default = { };
+      example = literalExpression ''
```

Comment:
Nit: you don't need `literalExpression` here.

Reply:
fixed, thanks!